test/prototype/models/test_models.py

```diff
@@ -0,0 +1,130 @@
+
+        # case: user provide encoder-decoder checkpoint state dict
+        dummy_t5_conf = T5Conf(
+            encoder_only=False,
+            vocab_size=10,
+            embedding_dim=16,
+            ffn_dimension=64,
+            num_attention_heads=2,
+            num_encoder_layers=2,
+        )
+        dummy_t5 = T5Model(dummy_t5_conf)
+        t5_model = T5Bundle.build_model(config=dummy_t5_conf, checkpoint=dummy_t5.state_dict())
+        self.assertEqual(t5_model.state_dict(), dummy_t5.state_dict())
+
+    @patch("logging.Logger.warning")
+    def test_t5_bundler_get_model(self, mock):
+        from torchtext.prototype.models import T5Conf, T5Bundle
+
+        # encoder-only
+        dummy_encoder_conf = T5Conf(
+            encoder_only=True,
+            vocab_size=10,
+            embedding_dim=16,
+            ffn_dimension=64,
+            num_attention_heads=2,
+            num_encoder_layers=2,
+        )
+        encoder_bundle = T5Bundle(dummy_encoder_conf)
+        encoder_bundle.get_model(load_weights=False, freeze_model=True)
+        mock.assert_called_with(
+            "The model is not loaded with pre-trained weights. Setting freeze_model to True will hinder model from learning appropriate weights."
+        )
+
+        # encoder-decoder
+        dummy_t5_conf = T5Conf(
+            encoder_only=False,
+            vocab_size=10,
+            embedding_dim=16,
+            ffn_dimension=64,
+            num_attention_heads=2,
+            num_encoder_layers=2,
+        )
+        t5_bundle = T5Bundle(dummy_t5_conf)
+        t5_bundle.get_model(load_weights=False, freeze_model=True)
+        mock.assert_called_with(
+            "The model is not loaded with pre-trained weights. Setting freeze_model to True will hinder model from learning appropriate weights."
+        )
+
+    def test_t5_bundler_raise_checkpoint(self):
+        from torchtext.prototype.models import T5Conf, T5Bundle
+
+        # encoder-only
+        with self.assertRaises(TypeError):
+            dummy_encoder_conf = T5Conf(
+                encoder_only=True,
+                vocab_size=10,
+                embedding_dim=16,
+                ffn_dimension=64,
+                num_attention_heads=2,
+                num_encoder_layers=2,
+            )
+            T5Bundle.build_model(
+                config=dummy_encoder_conf,
+                freeze_model=True,
+                checkpoint=1,
+            )
+
+        # encoder-decoder
+        with self.assertRaises(TypeError):
+            dummy_t5_conf = T5Conf(
+                encoder_only=False,
+                vocab_size=10,
+                embedding_dim=16,
+                ffn_dimension=64,
+                num_attention_heads=2,
+                num_encoder_layers=2,
+            )
+            T5Bundle.build_model(
+                config=dummy_t5_conf,
+                freeze_model=True,
+                checkpoint=1,
+            )
+
+    def test_t5_bundler_conf_property(self):
```

Comment:
nit: Perhaps we can test only with encoder-decoder conf, seems like testing both with encoder only and encoder-decoder conf could be redundant?

Reply:
sure! do you mean for all the tests (since the only difference between testing the encoder and encoder-decoder models is the value passed for `encoder-only` when initializing the configs), or just remove the redundancy for `test_t5_bundler_conf_property()`?

Reply:
right, so we could either test with encode-only or encoder-decoder conf, but doing with both might be redundant.

Reply:
And I  meant for tests within `test_t5_bundler_conf_property()` only.

Reply:
okay, i'll remove the one for encoder-only. thanks!